lgc/builder/BuilderImpl.cpp

```diff
@@ -446,3 +446,13 @@ Value *BuilderImplBase::scalarize(Value *value0, Value *value1, Value *value2,
   Value *result = callback(value0, value1, value2);
   return result;
 }
+
+// =====================================================================================================================
+// Create code to get the lane number within the wave. This depends on whether the shader is wave32 or wave64,
+// and thus on the shader stage it is used from.
+Value *BuilderImplBase::CreateGetLaneNumber() {
+  Value *result = CreateIntrinsic(Intrinsic::amdgcn_mbcnt_lo, {}, {getInt32(-1), getInt32(0)});
+  if (getPipelineState()->getShaderWaveSize(m_shaderStage) == 64)
+    result = CreateIntrinsic(Intrinsic::amdgcn_mbcnt_hi, {}, {getInt32(-1), result});
+  return result;
+}
```

Comment:
They can't use it directly, because this is in BuilderImplBase, not BuilderBase. And it has to be there because it needs to get the current shader's wave size. I guess the code could move to BuilderBase and take the wave size as an additional arg.

Reply:
I didn't make that change, on the basis that we will hopefully eventually move all the PatchInOutImportExport code for built-ins over to InOutBuilder.